#### UnionPeakSet for Th1, Th2, Th17 and Treg

In [10]:
%%bash
mkdir -p /data/reddylab/Revathy/collabs/Keith/Tcell
mkdir -p /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
in_dir=/data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200805A5-pe-blacklist-removal
outdir=/data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
files=(\
KS130.Th1.96hrs.ASTARR.insert.rep1.masked.dedup.sorted_peaks.narrowPeak \
KS130.Th1.96hrs.ASTARR.insert.rep2.masked.dedup.sorted_peaks.narrowPeak \
KS131.Th2.96hrs.ASTARR.insert.rep1.masked.dedup.sorted_peaks.narrowPeak \
KS131.Th2.96hrs.ASTARR.insert.rep2.masked.dedup.sorted_peaks.narrowPeak \
KS133.Th0.72hrs.ASTARR.insert.rep1.masked.dedup.sorted_peaks.narrowPeak \
KS133.Th0.72hrs.ASTARR.insert.rep2.masked.dedup.sorted_peaks.narrowPeak \
KS133.Treg.72hrs.ASTARR.insert.rep1.masked.dedup.sorted_peaks.narrowPeak \
KS133.Treg.72hrs.ASTARR.insert.rep2.masked.dedup.sorted_peaks.narrowPeak 
)
for file in ${files[@]}; 
do 
  ### sort the bed files
  sort -k1,1 -k2,2n -k3,3n ${in_dir}/${file} > ${outdir}/${file/.masked.dedup.sorted_peaks.narrowPeak/.sorted_peaks.narrowPeak}

done

In [9]:
%%bash
in_dir=/data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200804A5-pe-blacklist-removal/
outdir=/data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
files=(\
KS134.Th17.72hrs.ASTARR.insert.rep1.masked.dedup.sorted_peaks.narrowPeak \
KS134.Th17.72hrs.ASTARR.insert.rep2.masked.dedup.sorted_peaks.narrowPeak \
KS136.Th17.72hrs.ASTARR.insert.rep3.masked.dedup.sorted_peaks.narrowPeak \
KS136.Th17.72hrs.ASTARR.insert.rep4.masked.dedup.sorted_peaks.narrowPeak
)
for file in ${files[@]}; 
do 
  ### sort the bed files
  sort -k1,1 -k2,2n -k3,3n ${in_dir}/${file} > ${outdir}/${file/.masked.dedup.sorted_peaks.narrowPeak/.sorted_peaks.narrowPeak}

done

In [7]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
cat KS130.Th1.96hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS130.Th1.96hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
KS131.Th2.96hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS131.Th2.96hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
KS133.Treg.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS133.Treg.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
KS134.Th17.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS134.Th17.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
KS136.Th17.72hrs.ASTARR.insert.rep3.sorted_peaks.narrowPeak \
KS136.Th17.72hrs.ASTARR.insert.rep4.sorted_peaks.narrowPeak \
| sort -k1,1 -k2,2n | bedtools merge -i - > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak

In [8]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
cat KS133.Th0.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS133.Th0.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
| sort -k1,1 -k2,2n | bedtools merge -i - > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0.72hrs.UnionPeakSet.narrowPeak

In [16]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
cat KS130.Th1.96hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS130.Th1.96hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
| sort -k1,1 -k2,2n | bedtools merge -i - > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1.96hrs.UnionPeakSet.narrowPeak

In [17]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
cat KS131.Th2.96hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS131.Th2.96hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
| sort -k1,1 -k2,2n | bedtools merge -i - > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th2.96hrs.UnionPeakSet.narrowPeak

In [18]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
cat KS134.Th17.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS134.Th17.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak \
KS136.Th17.72hrs.ASTARR.insert.rep3.sorted_peaks.narrowPeak \
KS136.Th17.72hrs.ASTARR.insert.rep4.sorted_peaks.narrowPeak \
| sort -k1,1 -k2,2n | bedtools merge -i - > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th17.72hrs.UnionPeakSet.narrowPeak

In [25]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
cat KS133.Treg.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak \
KS133.Treg.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak\
| sort -k1,1 -k2,2n | bedtools merge -i - > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Treg.72hrs.UnionPeakSet.narrowPeak

In [19]:
%%bash
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
wc -l *.narrowPeak

    98875 KS130.Th1.96hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak
    91488 KS130.Th1.96hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak
    51557 KS131.Th2.96hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak
    60891 KS131.Th2.96hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak
    78104 KS133.Th0.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak
    74733 KS133.Th0.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak
    68903 KS133.Treg.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak
    78262 KS133.Treg.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak
    83991 KS134.Th17.72hrs.ASTARR.insert.rep1.sorted_peaks.narrowPeak
    85202 KS134.Th17.72hrs.ASTARR.insert.rep2.sorted_peaks.narrowPeak
    85375 KS136.Th17.72hrs.ASTARR.insert.rep3.sorted_peaks.narrowPeak
    84222 KS136.Th17.72hrs.ASTARR.insert.rep4.sorted_peaks.narrowPeak
   941603 total


In [21]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak

138911 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak


In [2]:
%%bash
source /data/reddylab/software/miniconda2/bin/activate alex
outdir=/data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq
python /data/reddylab/Alex/reddylab_utils/scripts/bed_to_saf.py \
    --bed6 \
    -beds \
        ${outdir}/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak \
    -safs \
        ${outdir}/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.saf

In [3]:
!head /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.saf

chr1_4491842_4493419	chr1	4491842	4493419	+
chr1_4496199_4497155	chr1	4496199	4497155	+
chr1_4571547_4572067	chr1	4571547	4572067	+
chr1_4600496_4601119	chr1	4600496	4601119	+
chr1_4671434_4672049	chr1	4671434	4672049	+
chr1_4689010_4689626	chr1	4689010	4689626	+
chr1_4724465_4725198	chr1	4724465	4725198	+
chr1_4745580_4745966	chr1	4745580	4745966	+
chr1_4747901_4748766	chr1	4747901	4748766	+
chr1_4769822_4770396	chr1	4769822	4770396	+


In [4]:
%%bash
mkdir -p /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/featureCounts
cd /data/reddylab/Revathy/collabs/Keith/Th1/data/atac_seq/featureCounts
sbatch -p all \
    --cpus-per-task 16 \
    --mem 32G \
    -o /data/reddylab/Revathy/collabs/Keith/Tcell/logs/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.featurecounts.out \
    <<'EOF'
#!/bin/bash
/data/reddylab/software/subread-1.4.6-p4-Linux-x86_64/bin/featureCounts \
    -T 16 \
    -F SAF \
    -p \
    -a /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.saf \
    -o /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/featureCounts/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.featurecounts.txt \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200805A5-pe-blacklist-removal/KS130.Th1.96hrs.ASTARR.insert.rep1.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200805A5-pe-blacklist-removal/KS130.Th1.96hrs.ASTARR.insert.rep2.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200805A5-pe-blacklist-removal/KS131.Th2.96hrs.ASTARR.insert.rep1.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200805A5-pe-blacklist-removal/KS131.Th2.96hrs.ASTARR.insert.rep2.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200804A5-pe-blacklist-removal/KS134.Th17.72hrs.ASTARR.insert.rep1.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200804A5-pe-blacklist-removal/KS134.Th17.72hrs.ASTARR.insert.rep2.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200804A5-pe-blacklist-removal/KS136.Th17.72hrs.ASTARR.insert.rep3.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200804A5-pe-blacklist-removal/KS136.Th17.72hrs.ASTARR.insert.rep4.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200805A5-pe-blacklist-removal/KS133.Treg.72hrs.ASTARR.insert.rep1.masked.dups_marked.bam \
       /data/reddylab/Keith/encode4_duke/processing/atac_seq/Siklenka_6420_200805A5-pe-blacklist-removal/KS133.Treg.72hrs.ASTARR.insert.rep2.masked.dups_marked.bam
EOF

Submitted batch job 26323924


#### intersection between unionpeaks of Th0 and Th1,Th2,Th17 and Treg

In [10]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/narrowPeak
bedtools intersect \
-a Th0.72hrs.UnionPeakSet.narrowPeak \
-b Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak \
> /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.bed

***** WARNING: File Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	13778	13946

***** WARNING: File Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	13778	13946



In [1]:
!head /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.bed

chr1	4491842	4492903
chr1	4496364	4497155
chr1	4571571	4572067
chr1	4600679	4601024
chr1	4671434	4672049
chr1	4689010	4689626
chr1	4724588	4724820
chr1	4747985	4748565
chr1	4774950	4775866
chr1	4780086	4780487


In [26]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/*.narrowPeak

   84760 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0.72hrs.UnionPeakSet.narrowPeak
  101113 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th17.72hrs.UnionPeakSet.narrowPeak
  104142 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1.96hrs.UnionPeakSet.narrowPeak
  138911 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th1_Th2_Th17_Treg.72hrs.UnionPeakSet.narrowPeak
   67132 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th2.96hrs.UnionPeakSet.narrowPeak
   82535 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Treg.72hrs.UnionPeakSet.narrowPeak
  578593 total


In [27]:
%%bash
mkdir -p /data/reddylab/Revathy/collabs/Keith/Tcell/logs
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq
sbatch -p all \
    --cpus-per-task 1 \
    --mem 8G \
    -o /data/reddylab/Revathy/collabs/Keith/Tcell/logs/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.closest_gene.out \
    <<'EOF'
#!/bin/bash
SAMPLE="Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.bed"

tail -n+2 ${SAMPLE} \
| cut -f1,2,3 \
| awk -vOFS="\t" '{print $1, $2, $3, $1"_"$2"_"$3}' \
| sort -k1,1 -k2,2n \
| bedtools closest \
    -nonamecheck \
    -D a \
    -a - \
    -b /data/reddylab/Revathy/collabs/Keith/data/annotation_files/gencode.vM19.basic.annotation.TSS.bed \
| awk -vOFS="\t" \
    '{if(last==$4){genes=$11","genes;}else{if(genes){print peak, genes, distance}peak=$4; genes=$11; distance=$13; last=peak}}END{if(last==$4){genes=$11","genes;print peak, genes, distance}}' \
> ${SAMPLE/.bed/.gencode_closest_gene.txt}

EOF

Submitted batch job 26192542


In [31]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.bed

80630 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.bed


In [30]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.gencode_closest_gene.txt


80629 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.gencode_closest_gene.txt


In [37]:
%%bash
awk '{ if ($3 >= -3000 && $3 <= 3000) print $1}' /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.gencode_closest_gene.txt | wc -l

16415


In [46]:
%%bash 
cat /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.gencode_closest_gene.txt \
| awk '{ if ($3 >= -3000 && $3 <= 3000) print $1}' | awk -F '_' -vOFS="\t" '{print $1, $2, $3}' > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.promoters.bed

In [51]:
%%bash 
cat /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.gencode_closest_gene.txt \
| awk '{print $1}' | awk -F '_' -vOFS="\t" '{print $1, $2, $3}' > /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1_Th2_Th17_Treg.atac_seq.intersected_peaks.all_peaks.bed


#### Number of peaks

* Number of peaks when Th0 intersected with unionPeakSet of Th1, Th2, Th17 and Treg: 80629
* Number of peaks in the promoter region when Th0 intersected with unionPeakSet of Th1, Th2, Th17 and Treg: 16415

### Intersection of Th0 with each of the T cell subtypes

In [54]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/
bedtools intersect \
-a Th0.72hrs.UnionPeakSet.narrowPeak \
-b Th1.96hrs.UnionPeakSet.narrowPeak \
> /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1.atac_seq.intersected_peaks.bed

***** WARNING: File Th1.96hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	15713	17710

***** WARNING: File Th1.96hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	15713	17710



In [56]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1.atac_seq.intersected_peaks.bed

71473 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1.atac_seq.intersected_peaks.bed


In [57]:
%%bash
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq
sbatch -p all \
    --cpus-per-task 1 \
    --mem 8G \
    -o /data/reddylab/Revathy/collabs/Keith/Tcell/logs/Th0_Th1.atac_seq.intersected_peaks.closest_gene.out \
    <<'EOF'
#!/bin/bash
SAMPLE="Th0_Th1.atac_seq.intersected_peaks.bed"

tail -n+2 ${SAMPLE} \
| cut -f1,2,3 \
| awk -vOFS="\t" '{print $1, $2, $3, $1"_"$2"_"$3}' \
| sort -k1,1 -k2,2n \
| bedtools closest \
    -nonamecheck \
    -D a \
    -a - \
    -b /data/reddylab/Revathy/collabs/Keith/data/annotation_files/gencode.vM19.basic.annotation.TSS.bed \
| awk -vOFS="\t" \
    '{if(last==$4){genes=$11","genes;}else{if(genes){print peak, genes, distance}peak=$4; genes=$11; distance=$13; last=peak}}END{if(last==$4){genes=$11","genes;print peak, genes, distance}}' \
> ${SAMPLE/.bed/.gencode_closest_gene.txt}

EOF

Submitted batch job 26192959


In [58]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1.atac_seq.intersected_peaks.gencode_closest_gene.txt

71472 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1.atac_seq.intersected_peaks.gencode_closest_gene.txt


In [59]:
%%bash
awk '{ if ($3 >= -3000 && $3 <= 3000) print $1}' /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th1.atac_seq.intersected_peaks.gencode_closest_gene.txt | wc -l

15784


In [1]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/
bedtools intersect \
-a Th0.72hrs.UnionPeakSet.narrowPeak \
-b Th2.96hrs.UnionPeakSet.narrowPeak \
> /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th2.atac_seq.intersected_peaks.bed

***** WARNING: File Th2.96hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	13778	13946

***** WARNING: File Th2.96hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	13778	13946



In [2]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th2.atac_seq.intersected_peaks.bed

53861 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th2.atac_seq.intersected_peaks.bed


In [5]:
%%bash
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq
sbatch -p all \
    --cpus-per-task 1 \
    --mem 8G \
    -o /data/reddylab/Revathy/collabs/Keith/Tcell/logs/Th0_Th2.atac_seq.intersected_peaks.closest_gene.out \
    <<'EOF'
#!/bin/bash
SAMPLE="Th0_Th2.atac_seq.intersected_peaks.bed"

tail -n+2 ${SAMPLE} \
| cut -f1,2,3 \
| awk -vOFS="\t" '{print $1, $2, $3, $1"_"$2"_"$3}' \
| sort -k1,1 -k2,2n \
| bedtools closest \
    -nonamecheck \
    -D a \
    -a - \
    -b /data/reddylab/Revathy/collabs/Keith/data/annotation_files/gencode.vM19.basic.annotation.TSS.bed \
| awk -vOFS="\t" \
    '{if(last==$4){genes=$11","genes;}else{if(genes){print peak, genes, distance}peak=$4; genes=$11; distance=$13; last=peak}}END{if(last==$4){genes=$11","genes;print peak, genes, distance}}' \
> ${SAMPLE/.bed/.gencode_closest_gene.txt}

EOF

Submitted batch job 26193181


In [6]:
%%bash
awk '{ if ($3 >= -3000 && $3 <= 3000) print $1}' /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th2.atac_seq.intersected_peaks.gencode_closest_gene.txt | wc -l

14760


In [7]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/
bedtools intersect \
-a Th0.72hrs.UnionPeakSet.narrowPeak \
-b Th17.72hrs.UnionPeakSet.narrowPeak \
> /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th17.atac_seq.intersected_peaks.bed

***** WARNING: File Th17.72hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	15706	17696

***** WARNING: File Th17.72hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	15706	17696



In [8]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th17.atac_seq.intersected_peaks.bed

72323 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th17.atac_seq.intersected_peaks.bed


In [13]:
%%bash
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq
sbatch -p all \
    --cpus-per-task 1 \
    --mem 8G \
    -o /data/reddylab/Revathy/collabs/Keith/Tcell/logs/Th0_Th17.atac_seq.intersected_peaks.closest_gene.out \
    <<'EOF'
#!/bin/bash
SAMPLE="Th0_Th17.atac_seq.intersected_peaks.bed"

tail -n+2 ${SAMPLE} \
| cut -f1,2,3 \
| awk -vOFS="\t" '{print $1, $2, $3, $1"_"$2"_"$3}' \
| sort -k1,1 -k2,2n \
| bedtools closest \
    -nonamecheck \
    -D a \
    -a - \
    -b /data/reddylab/Revathy/collabs/Keith/data/annotation_files/gencode.vM19.basic.annotation.TSS.bed \
| awk -vOFS="\t" \
    '{if(last==$4){genes=$11","genes;}else{if(genes){print peak, genes, distance}peak=$4; genes=$11; distance=$13; last=peak}}END{if(last==$4){genes=$11","genes;print peak, genes, distance}}' \
> ${SAMPLE/.bed/.gencode_closest_gene.txt}

EOF

Submitted batch job 26193235


In [14]:
%%bash
awk '{ if ($3 >= -3000 && $3 <= 3000) print $1}' /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Th17.atac_seq.intersected_peaks.gencode_closest_gene.txt | wc -l


15872


In [15]:
%%bash
module load bedtools2
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/
bedtools intersect \
-a Th0.72hrs.UnionPeakSet.narrowPeak \
-b Treg.72hrs.UnionPeakSet.narrowPeak \
> /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Treg.atac_seq.intersected_peaks.bed

***** WARNING: File Treg.72hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	15676	17541

***** WARNING: File Treg.72hrs.UnionPeakSet.narrowPeak has inconsistent naming convention for record:
GL456216.1	15676	17541



In [16]:
!wc -l /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Treg.atac_seq.intersected_peaks.bed

67032 /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Treg.atac_seq.intersected_peaks.bed


In [17]:
%%bash
cd /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq
sbatch -p all \
    --cpus-per-task 1 \
    --mem 8G \
    -o /data/reddylab/Revathy/collabs/Keith/Tcell/logs/Th0_Treg.atac_seq.intersected_peaks.closest_gene.out \
    <<'EOF'
#!/bin/bash
SAMPLE="Th0_Treg.atac_seq.intersected_peaks.bed"

tail -n+2 ${SAMPLE} \
| cut -f1,2,3 \
| awk -vOFS="\t" '{print $1, $2, $3, $1"_"$2"_"$3}' \
| sort -k1,1 -k2,2n \
| bedtools closest \
    -nonamecheck \
    -D a \
    -a - \
    -b /data/reddylab/Revathy/collabs/Keith/data/annotation_files/gencode.vM19.basic.annotation.TSS.bed \
| awk -vOFS="\t" \
    '{if(last==$4){genes=$11","genes;}else{if(genes){print peak, genes, distance}peak=$4; genes=$11; distance=$13; last=peak}}END{if(last==$4){genes=$11","genes;print peak, genes, distance}}' \
> ${SAMPLE/.bed/.gencode_closest_gene.txt}

EOF

Submitted batch job 26193268


In [18]:
%%bash
awk '{ if ($3 >= -3000 && $3 <= 3000) print $1}' /data/reddylab/Revathy/collabs/Keith/Tcell/data/atac_seq/Th0_Treg.atac_seq.intersected_peaks.gencode_closest_gene.txt | wc -l


15506
